qiskit/algorithms/evolvers/pvqd.py

```diff
@@ -0,0 +1,309 @@
+            """Evaluate the observables for the ansatz parameters ``theta``.
+
+            Args:
+                theta: The ansatz parameters.
+
+            Returns:
+                The observables evaluated at the ansatz parameters.
+            """
+            value_dict = dict(zip(ansatz_parameters, theta))
+            sampled = self._sampler.convert(converted, params=value_dict)
+            return sampled.eval()
+
+        return evaluate_observables
+
+    def evolve(self, evolution_problem: EvolutionProblem) -> EvolutionResult:
+        """
+        Args:
+            evolution_problem: The evolution problem containing the hamiltonian, total evolution
+                time and observables to evaluate.
+
+        Returns:
+            A result object containing the evolution information and evaluated observables.
+
+        Raises:
+            ValueError: If the evolution time is not positive or the timestep is too small.
+        """
+        time = evolution_problem.time
+        observables = evolution_problem.aux_operators
+        hamiltonian = evolution_problem.hamiltonian
+
+        if not 0 < self.timestep <= time:
+            raise ValueError(
+                f"The time step ({self.timestep}) must be larger than 0 and smaller equal "
+                f"the evolution time ({time})."
+            )
+
+        # get the function to evaluate the observables for a given set of ansatz parameters
+        evaluate_observables = self._get_observable_evaluator(observables)
+
+        observable_values = [evaluate_observables(self.initial_parameters)]
+        fidelities = [1]
+        times = [0]
+        parameters = [self.initial_parameters]
+
+        current_time = 0
+
+        if self.initial_guess is None:
+            initial_guess = np.random.random(self.initial_parameters.size) * 0.01
+        else:
+            initial_guess = self.initial_guess
+
+        while current_time < time:
+            # perform VQE to find the next parameters
+            next_parameters, fidelity = self.step(
+                hamiltonian, parameters[-1], self.timestep, initial_guess
+            )
+
+            # set initial guess to last parameter update
+            initial_guess = next_parameters - parameters[-1]
+
+            # store parameters
+            parameters.append(next_parameters)
+            fidelities.append(fidelity)
+            observable_values.append(evaluate_observables(next_parameters))
+
+            # increase time
+            current_time += self.timestep
+            times.append(current_time)
+
+        evolved_state = self.ansatz.bind_parameters(parameters[-1])
+
+        result = PVQDResult(evolved_state=evolved_state, aux_ops_evaluated=observable_values[-1])
+        result.times = times
+        result.parameters = parameters
+        result.fidelities = fidelities
+        result.estimated_error = np.prod(result.fidelities)
+        result.observables = np.asarray(observable_values)
```

Comment:
Can we set values in the constructor? If not, can we extract the fragment as a method?

Reply:
Added to the constructor in a361656